tests/test_examples.py

```diff
@@ -0,0 +1,46 @@
+import glob
+import os
+import shutil
+
+import pytest
+
+from rapids_dependency_file_generator.rapids_dependency_file_generator import (
+    main as dfg,
+)
+
+CURRENT_DIR = os.path.dirname(__file__)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def clean_actual_files():
+    for root, _, _ in os.walk("tests"):
+        if os.path.basename(root) == "actual":
+            shutil.rmtree(root)
+
+
+def test1():
+    test_dir = os.path.join(CURRENT_DIR, "examples", "test1")
+    expected_dir = os.path.join(test_dir, "output", "expected")
+    actual_dir = os.path.join(test_dir, "output", "actual")
+    dep_file_path = os.path.join(test_dir, "dependencies.yaml")
+
+    dfg(dep_file_path)
+
+    expected_file_set = {
+        os.path.relpath(f, expected_dir)
+        for f in glob.glob(expected_dir + "/**", recursive=True)
+        if os.path.isfile(f)
+    }
+    actual_file_set = {
+        os.path.relpath(f, actual_dir)
+        for f in glob.glob(actual_dir + "/**", recursive=True)
+        if os.path.isfile(f)
+    }
+
+    assert len(expected_file_set) == len(actual_file_set)
+    assert expected_file_set == actual_file_set
```

Comment:
The length check is kind of redundant -- and if pytest fails at that point, it won't be very helpful because you won't see which files are different. The second check of the sets is helpful and pytest will show nice output if they are not equal. Finally, we want to use the order `assert actual == expected` for the best error messages from pytest. 
```suggestion
    assert actual_file_set == expected_file_set
```

Example: 
![image](https://user-images.githubusercontent.com/3943761/197038354-c15e85bd-cca0-46e1-8a95-7081323dccbd.png)

The order affects how the diff between the two objects is displayed.